### ISSUE TITLE
Integrate OWASP DefectDojo as an alternative persistenceProvider

### ISSUE THREAD

**Issue by rseedorff** [secureCodeBox/secureCodeBox#50]
As a User i would like to use the [OWASP DefectDojo Projekt](https://www.owasp.org/index.php/OWASP_DefectDojo_Project) to analyse my findings (_instead or in parallel to my kibana dashboard_). While the secureCodeBox has its major focus on the automation part of the security scanner execution, defectDojo is good in visualising, deduplication and analysing the findings.

To integrate OWASP DefectDojo it's necessary to implement a new [persistenceProvider](https://github.com/secureCodeBox/engine/tree/master/scb-persistenceproviders) which pushes the raw finding results of each scanner to the [new DefectDojo API: ImportScan](https://app.swaggerhub.com/apis/DefectDojo/defect-dojo_api_v_2/1.0.0#/import-scan/import-scan_create). DefectDojo can be started as a separate Docker Container within the secureCodeBox Stack.

Possible Scanner Integrations:
- NMAP (XML output (use -oX))
- Nikto (XML output)
- ZAP (ZAP XML report format)
- Arachni (Arachni JSON report format.)

**Comment by rseedorff** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [DefectDojo Integration](https://trello.com/c/phiO03SV/165-defectdojo-integration)

**Comment by wurstbrot** [comment]
The database of defectdojo needs to be pre-filled with the following changes:
- User: Firstname, Lastname, Block execution
- System: Settings deduplication enabled, False Positive History enabled
- admin-Password trough env. variables

**Comment by rseedorff** [comment]
There are some changes regarding the container setup (docker and docker-compose) we are currently waiting to be merged and released... https://github.com/DefectDojo/django-DefectDojo/issues/921

**Comment by wurstbrot** [comment]
Is merged and is in release candidate 1.5.4rc3

**Comment by rseedorff** [comment]
Thx for this hint, so we are waiting for the latest release 1.5.4 :)

**Comment by wurstbrot** [comment]
It is in master now

**Comment by J12934** [comment]
Will work on integrating defectdojo into the docker-compose stack in the next couple of days

**Comment by fuhrmeistery** [comment]
The integration for DefectDojo for the V2 will be done in another issue
